src/audio/copier.c

```diff
@@ -786,6 +786,30 @@ static inline int apply_attenuation(struct comp_dev *dev, struct copier_data *cd
 		return -EINVAL;
 	case SOF_IPC_FRAME_S24_4LE:
 	case SOF_IPC_FRAME_S32_LE:
+#if __XCC__
```

Comment:
My preference is to have this logic in a copier_hifi.c file which can then check for the correct hifi versions (as we will likely need other SIMD versions too).

Reply:
actually in HiFi version we only use 2-ways load/store/shift, up to now, the hifi3 & hifi4 is the same, so I just have a c file named copier_hifi.c